README.md

```diff
@@ -73,8 +73,11 @@ const sw = Splitwise({
   consumerSecret: 'your secret here',
   grant_type: 'authorization_code',
   redirect_uri: 'your redirect_uri'
+  state: 'your state' // OPTIONAL. But we strongly recomend to have a state parameter and verify it later on as mentioned below 
```

Comment:
ah, this won't work, the state must be distinct for every attempted authentication

Reply:
oh yaa! I get it. so how about the client supplies a function that will generate state everytime? But do we expose this state to the client so that they can verify it? Because passing a static state will always result in having the same state for each auth request.

Reply:
I think the client should just supply the state individually for each request. We would have to expose it - not just so that they can verify it - it might be used to store information about how the user initiated the original request (such as which page to redirect back  @omkarsk98

Reply:
@keriwarr Got it. So this is my understanding. Correct me if I am wrong. we add a state key in `getAuthorizationUrl`. If that is present, we use that, else we generate our state and we expose `getState` for both the cases?

Reply:
That's what i was originally thinking, but i re-read the OAuth spec for the "state" parameter, and my new understanding is that in order to verify the state, it's insufficient to check that it matches the outgoing state, you actually have to check that the user for whom you generated the auth url is the same user who visited the callback. Our library isn't in a position to do that so we have to leave it to the client, meaning that we shouldn't generate a state, we should require the client to generate one. Moreover, the client will revive the state from the inbound callback request, so we shouldn't need to provide a getState mechanism. Indeed, we might not be able to as in theory there could be many users doing oauth authentication at the same time - how would we store all their different states and keep them disentangled without knowing who they are?